src/averageAges.js

```diff
@@ -37,7 +49,22 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenFilter = people
+    .filter(({ sex }) => sex === 'f');
+
+  const womenFilterWithChild = womenFilter
+    .filter(({ name }) => people.some(({ mother }) => mother === name));
+
+  const selectedWomen = withChildren ? womenFilterWithChild : womenFilter;
+  const ageDifferences = selectedWomen
+    .map(({ died, born }) => died - born);
+
+  const totalAgeDifference = ageDifferences
+    .reduce((acc, value) => acc + value);
+
+  const avgAgeOfWomen = totalAgeDifference / ageDifferences.length;
```

Comment:
Similar logic occurs in the previous function, so we can transfer these calculations to a separate function and use it as a helper.